### DISCUSSION TITLE
How to read customized extensionObjects

### DISCUSSION THREAD

**Discussion by vdkost** [gopcua/opcua#486] [Q&A]
what do I have to do to read more complex extensionObjects of customized dataType,
e.g.: typePotential as FloatCollection

![example_extensionObjectType_of_FloatCollection_of_Single](https://user-images.githubusercontent.com/61361394/137885336-834486f4-438a-436b-88c2-3918594208bf.PNG)

or type ECTA
![example_extensionObjectType_of_BooleanCollection_of_Boolean](https://user-images.githubusercontent.com/61361394/137885293-46c7deeb-c22f-44dd-9805-c5d67fed0443.PNG)
ct as BooleanCollection

currently I get the error message:
`invalid extension object with id ..`

**Comment by kung-foo** [comment]
You can register your custom type.

See examples in the generated code for well known extensions: https://github.com/gopcua/opcua/blob/main/ua/register_extobjs_gen.go

**Comment by vdkost** [comment]
Can you please provide me a code snippet for the above example. I am not familiar with go.
